ban-cooking.lua

```diff
@@ -0,0 +1,433 @@
+                type_list[key_matinfo_type]["subtype"] = -1
+
+            end
+
+            if m.reaction_product and m.reaction_product.id and reaction_product_id_contains(m.reaction_product, "SOAP_MAT") then
+                if not(type_list[key_matinfo_type]) then
+                    type_list[key_matinfo_type] = {}
+                end
+
+                type_list[key_matinfo_type]["text"] = c.name[1] .. " tallow"
+                -- item type for milk appears to use the 'GLOB' tag
+                type_list[key_matinfo_type]["type"] = "GLOB"
+                type_list[key_matinfo_type]["subtype"] = -1
+            end
+        end
+    end
+
+    local output = {}
+
+    for i, b in pairs(already_banned) do
+        -- initialize our output string with the array entry position (largely stays the same for each item on successive runs, except when items are added/removed)
+        local cur = ''
+        -- initialize our key for accessing our stored items info
+        local key = b.mat_type .. b.mat_index
+
+        -- It shouldn't be possible for there to not be a matching key entry by this point, but we'll be kinda safe here
+        if type_list[key] then
+            -- Add the item name to the first part of the string
+            cur = type_list[key]['text'] .. ' |type '
+
+            if type_list[key]['type'] == df.item_type[b.type] then
+                cur = cur .. 'match: ' .. type_list[key]['type']
+            else
+                -- Aw crap.  The item type we EXpected doesn't match up with the ACtual item type.
+                cur = cur .. "error: ex;" .. type_list[key]["type"] .. "/ac;" .. df.item_type[b.type]
+            end
+
+            cur = cur .. "|subtype "
+            if type_list[key]["subtype"] == b.subtype then
+                -- item sub type is a match, so we print that it's a match, as well as the item subtype index number (-1 means there is no subtype for this item)
+                cur = cur .. "match: " .. type_list[key]["subtype"]
+            else
+                -- Something went wrong, and the EXpected item subtype index value doesn't match the ACtual index value
+                cur = cur .. "error: ex;" .. type_list[key]["subtype"] .. "/ac;" .. b.subtype
+            end
+        else
+            -- There's no entry for this item in our calculated list of cookable items.  So, it's not a plant, alcohol, tallow, or milk.  It's likely that it's a meat that has been banned.
+            cur = cur .. '|"' .. '[' .. b.mat_type .. ', ' .. b.mat_index .. ']' .. ' unknown banned material type (meat?) " ' .. '|item type: "' .. tostring(df.item_type[b.type]) .. '"|item subtype: "' .. tostring(b.subtype)
+        end
+
+        table.insert(output, cur)
+    end
+
+    table.sort(output, function(a, b) return a < b end)
+
+    for k, v in pairs(output) do
+        print(k .. ": |" .. v)
+    end
+end
+
+if ... == "help" then
+    print(dfhack.script_help())
```

Comment:
needs a `return` statement, otherwise the script will error out after printing the help